modules/core/src/test/java/org/apache/ignite/internal/util/IgniteUtilsSelfTest.java

```diff
@@ -1296,6 +1303,95 @@ public void testDoInParallelException() {
         }
     }
 
+    /**
+     *
+     */
+    @Test
+    public void testGridConcurrentMultiPairQueueCorrectness() throws Exception {
```

Comment:
This test contains two different scenarios, can you separate it accordingly, please. Also, I think it makes sense to extract this test to their personal class like GridConcurrentMultiPairQueueTest

Reply:
yep, done.